pallets/asset-manager/src/lib.rs

```diff
@@ -118,19 +123,27 @@ pub mod pallet {
 	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
 	pub enum Event<T: Config> {
 		AssetRegistered(T::AssetId, T::AssetType, T::AssetRegistrarMetadata),
-		UnitsPerSecondChanged(T::AssetId, u128),
+		UnitsPerSecondChanged(T::AssetType, u128),
+		AssetTypeChanged(T::AssetId, T::AssetType),
 	}
 
-	/// Stores the asset TYPE
+	/// Stores the asset Type with assetId as key
 	#[pallet::storage]
 	#[pallet::getter(fn asset_id_type)]
 	pub type AssetIdType<T: Config> = StorageMap<_, Blake2_128Concat, T::AssetId, T::AssetType>;
 
+	/// Stores the asset Id with Asset Type as key
+	#[pallet::storage]
+	#[pallet::getter(fn asset_type_id)]
+	pub type AssetTypeId<T: Config> = StorageMap<_, Blake2_128Concat, T::AssetType, T::AssetId>;
+
 	// Stores the units per second for local execution.
 	// Not all assets might contain units per second, hence the different storage
+	// AssetType as key
```

Comment:
This seems implied by the type, are you perhaps trying to document something else here?

Reply:
probably it was more a mental note than anything else :). I will remove it